Source/Urho3D/CSharp/Managed/Scene/Node.cs

```diff
@@ -34,9 +38,68 @@ public partial class Node
             return (T)GetComponent(typeof(T).Name, recursive);
         }
 
+        /// <summary>
+        /// Get first occurance of a component type
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <returns></returns>
         public T GetComponent<T>() where T : Component
         {
             return (T)GetComponent(typeof(T).Name);
         }
+
+        /// <summary>
+        /// get all components of a type
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <returns></returns>
+        public ComponentList GetComponentsbyType<T>(bool recursive = false) where T : Component
+        {
+            ComponentList cplist = new ComponentList();
+            GetComponents(cplist, typeof(T).Name, recursive);
+            return cplist;
+        }
+
+        /// <summary>
+        /// Remove all Components of a specific Type
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        public void RemoveAllComponents<T>() where T : Component
+        {
+            var childrenNode = GetComponentsbyType<T>();
+            foreach (var childnode in childrenNode)
+            {
+                RemoveComponent(childnode);
+            }
+            Debug.Assert(GetComponent<T>() is null, "Components are not removed");
```

Comment:
I dont think this assert is needed. Besides it does not validate what current implementation does.

Implementation is also inefficient. We could simply do `RemoveComponents(typeof(T).Name)`.

`childrenNode` is misleading. Variable stores component of certain type. This has nothing to do with either children or nodes.

All in all this should simply be:
```cs
public void RemoveComponents<T>() where T : Component
{
    RemoveComponents(typeof(T).Name);
}
```

Reply:
yep I agree. not required at all. I will omit it